### PR TITLE
PROCESS API enhancement: required input values listed for model selections

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

Closes #{ID}

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `pre-commit run --from-ref develop --to-ref HEAD`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: eudemo/eudemo/radial_build.py [hunk 1]
```diff
@@ -218,7 +218,7 @@
         "alstroh": 6.6e8,
         "psepbqarmax": 9.2,
         "aspect": 3.1,
-        "m_s_limit": 0.1,
+        # "m_s_limit": 0.1,
```
Comment (je-cook) [inline eudemo/eudemo/radial_build.py:221]: for my own info, why is this commented?

**Review by je-cook — CHANGES_REQUESTED** [review]
Bit unsure about the enum <-> dataclass interplay but if it works I can't think of a nicer way off the top of my head that doesnt require some extra initialisation step.

Otherwise immutability changes again, I'll have a bit more of a think while that happens.

File: bluemira/codes/process/_equation_variable_mapping.py [hunk 2]
```diff
@@ -21,6 +21,230 @@
...
+        2,
+        [1, 2, 3, 4, 6, 10, 11],
+        [],
+        "Global Power Balance Consistency",
+    )
+    ION_POWER_CONSISTENCY = (
+        3,
+        [1, 2, 3, 4, 6, 10, 11],
+        [],
+        "DEPRECATED - Ion Power Balance Consistency",
```
[the remark this answers is not in the document]
Comment (CoronelBuendia) [reply bluemira/codes/process/_equation_variable_mapping.py:69]: The problem with some of these constraints is that depending on where you look in the PROCESS documentation, they are either deprecated / not used, or .. well, not. I will try and run these ones down in the source code and see if there is more clarity there.

File: eudemo/eudemo/radial_build.py (hunk 1 above)
Comment (CoronelBuendia) [reply eudemo/eudemo/radial_build.py:221]: Sorry, this was just to test that a warning came up, will put it back

**Review by je-cook — APPROVED** [review]
Yeah looks good just testing now, then we can start using it

File: bluemira/codes/process/_setup.py [hunk 3]
```diff
@@ -104,14 +104,14 @@ def _write_in_dat(self, use_bp_inputs: bool = True):
...
-            for key, value in inputs.items():
-                writer.add_parameter(key, value)
-            for key, value in self.problem_settings.items():
-                writer.add_parameter(key, value)
+        # if use_bp_inputs:
+        #     inputs = self._get_new_inputs(remapper=update_obsolete_vars)
+        #     for key, value in inputs.items():
+        #         writer.add_parameter(key, value)
+        #     for key, value in self.problem_settings.items():
+        #         writer.add_parameter(key, value)
```
Comment (CoronelBuendia) [inline bluemira/codes/process/_setup.py:112]: @je-cook I needed to do this to play around, as discussed, but results in errors after PROCESS run in EU-DEMO.

Comment (je-cook) [reply bluemira/codes/process/_setup.py:112]: no problem, I'll rebase mine and see if I can find the root.

File: bluemira/codes/process/_inputs.py [hunk 4]
```diff
@@ -315,7 +716,7 @@ def to_invariable(self) -> Dict[str, _INVariable]:
         """
         out_dict = {}
         for name, value in self:
-            if name not in ["icc", "ixc", "bounds"]:
+            if name not in ["icc", "ixc", "bounds"] and value is not None:
                 new_val = _INVariable(name, value, "Parameter", "", "")
                 out_dict[name] = new_val
         out_dict["icc"] = _INVariable(
```
Comment (je-cook) [inline bluemira/codes/process/_inputs.py:722]: might need to do something similar with icc ixc and bounds if they're None. Not important right now but jsut flagging

**Review by je-cook — APPROVED** [review]

**Pull request merged**